tests/CMakeLists.txt

```diff
@@ -79,4 +79,35 @@ add_executable(knnepsilon knnepsilon.cpp)
 target_link_libraries(knnepsilon ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
 
 add_executable(knnbucketsize knnbucketsize.cpp)
-target_link_libraries(knnbucketsize ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
\ No newline at end of file
+target_link_libraries(knnbucketsize ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
+
+# Ensure that users cannot instantiate a tree with wrong matrix types.
+macro(try_compile_disallowed_types MAT_TYPE TREE_TYPE EXPECT)
```

Comment:
The ```EXPECT``` variable is in conflict with ```disallowed``` here somehow, I believe. Maybe something ```try_compile_cloud_types``` would be more accurate?